scripts/control_bar.js

```diff
@@ -2,74 +2,149 @@
 /* exported makeImageWidget */
 
 // Construct the image sizing controls.
-var makeImageControl = function(imageElement) {
+var makeImageControl = function(content) {
+    const imageCursor = "grab";
+    // use plain js image so width or style.width is clearly differentiated
+    const image = document.createElement("img");
+    image.classList.add("middle-align");
+    image.style.cursor = imageCursor;
+    // When the image is rotated it has width and height as if it were not
+    // rotated. To make scroll work correctly, enclose it in a div with the
+    //actual width and height.
+    const imageDiv = document.createElement("div");
+    imageDiv.classList.add("middle-align", "left-align");
+    imageDiv.style.overflow = "hidden";
+    imageDiv.style.display = "inline-block";
+    content.append(imageDiv);
+    imageDiv.appendChild(image);
+
+    let scrollDiffX = 0;
+    let scrollDiffY = 0;
+    function mousemove(event) {
+        content.scrollTop(scrollDiffY - event.pageY);
+        content.scrollLeft(scrollDiffX - event.pageX);
+    }
+
+    function mouseup() {
+        $(document).unbind("mousemove mouseup");
+        image.style.cursor = imageCursor;
+    }
+
+    $(image).mousedown( function(event) {
+        event.preventDefault();
+        image.style.cursor = "grabbing";
+        scrollDiffX = event.pageX + content.scrollLeft();
+        scrollDiffY = event.pageY + content.scrollTop();
+        $(document).on("mousemove", mousemove)
+            .on("mouseup", mouseup);
+    });
+
     let imageKey;
     // percent need not be an integer but is rounded for display and save
     // it will typically not be an integer after fit height or width or + or -
     let percent;
     const minPercent = 10;
     const maxPercent = 999;
     const defaultPercent = 100;
+    let sine = 0;
```

Comment:
While I understand these are ultimately the values used in the CSS, it would be much more intuitive in the code if we stored a clockwise rotation angle (0, 90, 180, 270) and converted that into sine/cosine in `setImageStyle()` via a lookup table.

Reply:
We could do that but I think this method is simpler and avoids having to reset the index if the user rotates through more than 360.

Reply:
That's fine. Please add a block comment here that maps the sine/cosine values to their rotation angle to help the next person who reads the code. Something like:
```js
// sine | cosine | clockwise rotation
// ----------------------------------
//   x  |   y    | z degrees
//   x  |   y    | z degrees
//   x  |   y    | z degrees
//   x  |   y    | z degrees
```

Reply:
Good idea, I'll do that.